src/Common/src/System/IO/StringParser.cs

```diff
@@ -94,6 +94,37 @@ public string MoveAndExtractNext()
         }
 
         /// <summary>
+        /// Moves to the next component of the string, which must be enclosed in the only set of top-level parentheses
+        /// in the string.  The extracted value will be everything between (not including) those parentheses.
+        /// </summary>
+        /// <returns></returns>
+        public string MoveAndExtractNextInOuterParens()
+        {
+            // Move to the next position
+            MoveNextOrFail();
+
+            // After doing so, we should be sitting at a the opening paren.
+            if (_buffer[_startIndex] != '(')
+            {
+                ThrowForInvalidData();
+            }
+
+            // Since we only allow for one top-level set of parentheses, find the last
+            // parenthesis in the string; it's paired with the opening one we just found.
+            int lastParen = _buffer.LastIndexOf(')');
```

Comment:
is it worth asserting that there is exactly one ) and no other ( in the string? presumably not